DALOptimizer/MatchInvocation.cs

```diff
@@ -0,0 +1,157 @@
+                    continue;
+                }
+                //catch clause
+                if (invocation.GetType().Name == CATCHCLAUSE)
+                {
+                    MatchCatchClause(invocation, file, astResolver);
+                    continue;
+                }
+
+                // For All Global Field Declarations
+                if (invocation.GetType().Name == FIELDDECLARATION)
+                {
+                    MatchFieldDecl(invocation, file, astResolver);
+                    continue;
+                }
+                // For variable Decaration of type {SqlCommand cmd = new SqlCommand ("dbo.InboxDeviceReport", con);}
+                if (invocation.GetType().Name == VARIABLEDECLSTMT)
+                {
+                    MatchVarDeclStmt(invocation, file, astResolver);
+                    continue;
+                }
+
+                if (invocation.GetType().Name == ASSIGNMENTEXPRESSION)
+                {
+                    MatchAssExpr(invocation, file, astResolver);
+                    continue;
+                }
+
+                //check Global Property Declaration
+                if (invocation.GetType().Name == PROPERTYDECLARATION)
+                {
+                    MatchPropDecl(invocation, file, astResolver);
+                    continue;
+                }
+
+                //check Finally Block
+                if (invocation.GetType().Name == BLOCKSTATEMENT)
+                {
+                    MatchBlock(invocation, file, astResolver);
+                    continue;
+                }
+
+                if (invocation.GetType().Name == METHODDECLARARION)
+                {
+                    MatchMethodDecl(invocation, file, astResolver);
+                    continue;
+                }
+            }
+        }
+
+        // For All Global field Declarations eg. PrintFunction PrFun = new PrintFunction();
+        public void MatchFieldDecl(AstNode invocation, CSharpFile file, CSharpAstResolver astResolver)
+        {
+            if (invocation.Parent.GetType().Name.ToString() == TYPEDECLARATION)
+                file.IndexOfFieldDecl.Add((FieldDeclaration)invocation);
+        }
+
+        // For All Global Property Declarations
+        public void MatchPropDecl(AstNode invocation, CSharpFile file, CSharpAstResolver astResolver)
+        {
+            if (invocation.Parent.GetType().Name.ToString() == TYPEDECLARATION)
+                file.IndexOfPropDecl.Add((PropertyDeclaration)invocation);
+        }
+
+        public void MatchCatchClause(AstNode invocation, CSharpFile file, CSharpAstResolver astResolver)
+        {
+            file.IndexOfCtchClause.Add((CatchClause)invocation);
+        }
+
+        public void MatchAssExpr(AstNode invocation, CSharpFile file, CSharpAstResolver astResolver)
+        {
+            Match sqlCmdstmt1 = Pat.sqlCmdstmt1().Match(invocation);
+            if (sqlCmdstmt1.Success)
```

Comment:
@sushantyelpale rename `sqlCmdstmt1` to make it more meaningful. Variables should be named which we can pronounce.